include/swift/Parse/Lexer.h

```diff
@@ -24,6 +24,8 @@
 #include "llvm/ADT/SmallVector.h"
 
 namespace swift {
+  uint32_t validateUTF8CharacterAndAdvance(const char *&Ptr, const char *End);
+  bool EncodeToUTF8(unsigned CharValue, SmallVectorImpl<char> &Result);
```

Comment:
Sorry, I'm not sure what you're referring to below? I agree that I'd prefer not to publish them, but unfortunately I currently need `validateUTF8CharacterAndAdvance()` to iterate through the identifier in `TypeCheckConstraints.cpp`.